seed/seed20_test.go

```diff
@@ -162,6 +166,366 @@ func (s *seed20Suite) TestLoadMetaCore20Minimal(c *C) {
 	c.Check(runSnaps, HasLen, 0)
 }
 
+func (s *seed20Suite) makeCore20MinimalSeed(c *C, sysLabel string) string {
+	s.makeSnap(c, "snapd", "")
+	s.makeSnap(c, "core20", "")
+	s.makeSnap(c, "pc-kernel=20", "")
+	s.makeSnap(c, "pc=20", "")
+
+	s.MakeSeed(c, sysLabel, "my-brand", "my-model", map[string]interface{}{
+		"display-name": "my model",
+		"architecture": "amd64",
+		"base":         "core20",
+		"snaps": []interface{}{
+			map[string]interface{}{
+				"name":            "pc-kernel",
+				"id":              s.AssertedSnapID("pc-kernel"),
+				"type":            "kernel",
+				"default-channel": "20",
+			},
+			map[string]interface{}{
+				"name":            "pc",
+				"id":              s.AssertedSnapID("pc"),
+				"type":            "gadget",
+				"default-channel": "20",
+			}},
+	})
+
+	return filepath.Join(s.SeedDir, "systems", sysLabel)
+}
+
+func (s *seed20Suite) TestLoadAssertionsModelTempDBHappy(c *C) {
+	r := seed.MockTrusted(s.StoreSigning.Trusted)
+	defer r()
+
+	sysLabel := "20191031"
+	s.makeCore20MinimalSeed(c, sysLabel)
+
+	seed20, err := seed.Open(s.SeedDir, sysLabel)
+	c.Assert(err, IsNil)
+
+	err = seed20.LoadAssertions(nil, nil)
+	c.Assert(err, IsNil)
+
+	model, err := seed20.Model()
+	c.Assert(err, IsNil)
+	c.Check(model.Model(), Equals, "my-model")
+	c.Check(model.Base(), Equals, "core20")
+}
+
+func (s *seed20Suite) TestLoadAssertionsMultiModels(c *C) {
+	sysLabel := "20191031"
+	sysDir := s.makeCore20MinimalSeed(c, sysLabel)
+
+	err := osutil.CopyFile(filepath.Join(sysDir, "model"), filepath.Join(sysDir, "assertions", "model2"), 0)
+	c.Assert(err, IsNil)
+
+	seed20, err := seed.Open(s.SeedDir, sysLabel)
+	c.Assert(err, IsNil)
+
+	err = seed20.LoadAssertions(s.db, s.commitTo)
+	c.Check(err, ErrorMatches, `system cannot have any model assertion but the one in the system model assertion file`)
+}
+
+func (s *seed20Suite) TestLoadAssertionsInvalidModelAssertFile(c *C) {
+	sysLabel := "20191031"
+	sysDir := s.makeCore20MinimalSeed(c, sysLabel)
+
+	modelAssertFn := filepath.Join(sysDir, "model")
+
+	// copy over multiple assertions
+	err := osutil.CopyFile(filepath.Join(sysDir, "assertions", "model-etc"), modelAssertFn, osutil.CopyFlagOverwrite)
+	c.Assert(err, IsNil)
+
+	seed20, err := seed.Open(s.SeedDir, sysLabel)
+	c.Assert(err, IsNil)
+	err = seed20.LoadAssertions(s.db, s.commitTo)
+	c.Check(err, ErrorMatches, `system model assertion file must contain exactly the model assertion`)
+
+	// write whatever single non model assertion
+	seedtest.WriteAssertions(modelAssertFn, s.AssertedSnapRevision("snapd"))
+
+	seed20, err = seed.Open(s.SeedDir, sysLabel)
+	c.Assert(err, IsNil)
+	err = seed20.LoadAssertions(s.db, s.commitTo)
+	c.Check(err, ErrorMatches, `system model assertion file must contain exactly the model assertion`)
+}
+
+func (s *seed20Suite) massageAssertions(c *C, fn string, filter func(asserts.Assertion) asserts.Assertion) {
```

Comment:
Nice helper and nice use of the helper (below)!

Maybe "filterAssertions" or "writeFilteredAssertions" or "alterAssertions", "modifyAssertions"? Not super happy about this name.

Reply:
`writeFilteredAssertions` sounds like a good candidate for a name.